src/software/ai/hl/stp/tactic/shadow_enemy/shadow_enemy_fsm.h

```diff
@@ -0,0 +1,219 @@
+                DribblerMode::OFF, BallCollisionType::AVOID,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        /**
+         * Action to block the shot from our shadowee
+         *
+         *
+         * @param event ShadowEnemyFSM::Update
+         */
+        const auto block_shot = [](auto event) {
+            auto ball_position = event.common.world.ball().position();
+            auto face_ball_orientation =
+                (ball_position - event.common.robot.position()).orientation();
+            Point position_to_block = findBlockShotPoint(
+                event.common.robot, event.common.world.field(),
+                event.common.world.friendlyTeam(), event.common.world.enemyTeam(),
+                event.control_params.enemy_threat.value().robot,
+                event.control_params.shadow_distance);
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), position_to_block, face_ball_orientation, 0,
+                DribblerMode::OFF, BallCollisionType::AVOID,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        /**
+         * Action to steal and chip the ball
+         *
+         * Steal the ball if enemy threat is close enough and chip the ball away
+         *
+         * @param event ShadowEnemyFSM::Update
+         */
+        const auto steal_and_chip = [](auto event) {
+            auto ball_position = event.common.world.ball().position();
+            auto face_ball_orientation =
+                (ball_position - event.common.robot.position()).orientation();
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), ball_position, face_ball_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::AUTOCHIP, YEET_CHIP_DISTANCE_METERS},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *block_shot_s + update_e[!enemy_threat_has_ball] / block_pass = block_pass_s,
+            block_shot_s + update_e[is_blocking_shot] / steal_and_chip = steal_and_chip_s,
+            block_shot_s + update_e[!is_blocking_shot] / block_shot,
+            block_pass_s + update_e[!enemy_threat_has_ball] / block_pass,
+            block_pass_s + update_e[enemy_threat_has_ball] / block_shot = block_shot_s,
+            steal_and_chip_s + update_e[enemy_threat_has_ball] / steal_and_chip,
+            steal_and_chip_s + update_e[!enemy_threat_has_ball] / block_pass = X,
+            X + update_e[!enemy_threat_has_ball] / block_pass = block_pass_s,
+            X + update_e[enemy_threat_has_ball] / block_shot  = block_shot_s);
```

Comment:
The terminal state is actually redundant, as if we change line 215 to `steal_and_chip_s + update_e[!enemy_threat_has_ball] / block_pass = block_pass`, it essentially does the same thing. I only added the terminal state because the fsm needs to have a terminal state.